### PR TITLE
ycmd: 2017-11-05 -> 2018-06-14

### PR DESCRIPTION
**Pull request by LnL7**
Fixes build with boost 1.67

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested with vim/youcompleteme
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/tools/misc/ycmd/default.nix [hunk 1]
```diff
@@ -24,8 +24,6 @@ stdenv.mkDerivation rec {
     ${python.interpreter} build.py --system-libclang --clang-completer --system-boost
   '';
 
-  patches = [ ./dont-symlink-clang.patch ];
```
Comment (LnL7) [inline pkgs/development/tools/misc/ycmd/default.nix]: From what I can tell the patch isn't necessary anymore.

File: pkgs/development/tools/misc/ycmd/default.nix [hunk 2]
```diff
@@ -7,12 +7,12 @@
...
 stdenv.mkDerivation rec {
   name = "ycmd-${version}";
-  version = "2017-11-05";
+  version = "2018-06-14";
 
   src = fetchgit {
-    url = "git://github.com/Valloric/ycmd.git";
-    rev = "0ca16e7ba3a9078b697befdce64104e61829aa3b";
-    sha256 = "0bs94iv521ac2n53n3k8mw3s6v0hi3hhxhjsr0ips3n99al8wndi";
+    url = "https://github.com/Valloric/ycmd.git";
```
Comment (bhipple) [inline pkgs/development/tools/misc/ycmd/default.nix]: Thank you for swapping to HTTPS!  A lot of corporate networks and firewalls block SSH and FTP traffic from third party sites, but let HTTPS through; I was going to submit a PR to swap the url for `ycmd` but as always the wonderful Nix community is one step ahead :+1:

**Comment by bhipple** [comment]
LGTM

**Comment by xeji** [comment]
@GrahamcOfBorg  build vimPlugins.youcompleteme

**Pull request merged**